src/dataDisplay/icon/icon.jsx

```diff
@@ -2994,27 +3131,21 @@ class Icon extends React.PureComponent {
-                            {renderGradientColor}
-                        </defs>
-                        <g fill="none" fillRule="evenodd">
-                            {renderGradientColor ? (
-                                <mask id={maskId} fill="white">
-                                    <use xlinkHref={`#${pathId}`} />
-                                </mask>
-                            ) : null}
-                            <use className="icon-use-path" xlinkHref={`#${pathId}`} />
-                            {renderGradientColor ? (
-                                <rect fill={`url(#${gradientId})`} mask={`url(#${maskId})`} x="0" y="0" height="24" width="24" />
-                            ) : null}
-                        </g>
-                    </svg>
+                    <IconSVG
+                        gradientId={gradientId}
+                        maskId={maskId}
+                        pathD="M4.89851989,0.00195306941 C7.63309477,0.118914958 9.05094991,1.71205263 9.11877545,5.17823252 C9.18660098,8.64441242 6.41374236,9.66821441 6.55647743,12.0615829 C6.6992125,14.4549513 7.48184382,14.3132644 7.53617961,15.916259 C7.53075357,16.3837211 7.28341645,18.9518165 4.67243478,19 C1.12832445,18.9118931 1.26533203,16.0594327 1.28115799,16.0263927 C1.64982746,12.7193008 0.753626047,11.1254472 0.753626047,11.1254472 C0.753626047,11.1254472 -0.00300548474,9.09898888 8.98351072e-06,6.99543701 C0.270708233,-0.311266904 4.89851989,0.00195306941 4.89851989,0.00195306941 Z M19.1014801,5.00195307 C19.1014801,5.00195307 23.7292918,4.6887331 23.999991,11.995437 C24.0030055,14.0989889 23.246374,16.1254472 23.246374,16.1254472 C23.246374,16.1254472 22.3501725,17.7193008 22.718842,21.0263927 C22.734668,21.0594327 22.8716755,23.9118931 19.3275652,24 C16.7165835,23.9518165 16.4692464,21.3837211 16.4638204,20.916259 C16.5181562,19.3132644 17.3007875,19.4549513 17.4435226,17.0615829 C17.5862576,14.6682144 14.813399,13.6444124 14.8812246,10.1782325 C14.9490501,6.71205263 16.3669052,5.11891496 19.1014801,5.00195307 Z"
+                        pathId={pathId}
+                        renderGradientColor={renderGradientColor}
+                        style={svgStyle}
+                        title={title}
+                        type={type}
+                        viewBox="0 0 24 24"
+                    />
```

Comment:
I wonder if there is some better pattern that we could apply here.  Like a factory or builder type pattern.  Just musing aloud ... NAB for sure.  Let's move along and unblock things!